xurls.go

```diff
@@ -28,23 +28,23 @@ const (
 	iPrivateChar        = `\x{E000}-\x{F8FF}\x{F0000}-\x{FFFFD}\x{100000}-\x{10FFFD}`
 	midIChar            = `/?#\\` + midIPathSegmentChar + iPrivateChar
 	endIChar            = `/#` + endIPathSegmentChar + iPrivateChar
-	wellParen           = `\(([` + midIChar + `]|\([` + midIChar + `]*\))*\)`
-	wellBrack           = `\[([` + midIChar + `]|\[[` + midIChar + `]*\])*\]`
-	wellBrace           = `\{([` + midIChar + `]|\{[` + midIChar + `]*\})*\}`
+	wellParen           = `\((?:[` + midIChar + `]|\([` + midIChar + `]*\))*\)`
+	wellBrack           = `\[(?:[` + midIChar + `]|\[[` + midIChar + `]*\])*\]`
+	wellBrace           = `\{(?:[` + midIChar + `]|\{[` + midIChar + `]*\})*\}`
 	wellAll             = wellParen + `|` + wellBrack + `|` + wellBrace
-	pathCont            = `([` + midIChar + `]*(` + wellAll + `|[` + endIChar + `]))+`
+	pathCont            = `(?:[` + midIChar + `]*(?:` + wellAll + `|[` + endIChar + `]))+`
```

Comment:
I hadn't thought that capturing groups incur overhead even when they're not used in practice. Makes sense in hindsight.